ci/infra/libvirt/cloud-init/common.tpl

```diff
@@ -34,17 +34,20 @@ ${repositories}
     download.use_deltarpm: "true"
 
 # WARNING!!! Do not use cloud-init packages module when SUSE CaaSP Registraion
-# Code is provided. In this case repositories will be added in runcmd module 
+# Code is provided. In this case repositories will be added in runcmd module
 # with SUSEConnect command after packages module is ran
 #packages:
 
+# set hostname
+hostname: ${hostname}
+
 bootcmd:
   - ip link set dev eth0 mtu 1400
-  # Hostnames from DHCP - otherwise localhost will be used
-  - /usr/bin/sed -ie "s#DHCLIENT_SET_HOSTNAME=\"no\"#DHCLIENT_SET_HOSTNAME=\"yes\"#" /etc/sysconfig/network/dhcp
-  - netconfig update -f
 
 runcmd:
+  # Set node's hostname from DHCP server
+  - sed -i -e '/^DHCLIENT_SET_HOSTNAME/s/^.*$/DHCLIENT_SET_HOSTNAME=\"${hostname_from_dhcp}\"/' /etc/sysconfig/network/dhcp
+  - systemctl restart wicked
```

Comment:
Looks great to me restarting wicked rather than calling netconfig directly, this way we do not interfere with the network-online.target (I am quite convinced this `netconfig update -f` command is a source of issues, however I could not verify it in my tests). In any case, it is not clear to me why this has been moved from bootcmd to runcmd. I would expect the network configuration to happen completely at boot time, before the network-online.target is reached. For that the configuration should happen at bootcmd, is restarting wicked at bootcmd causing issues?

Reply:
Thanks for giving me that `runcmd` is at post network-online.target.

The main reason I did is due to the cloud-init doc:
> bootcmd will run on every boot
> runcmd only runs during the first boot

But I agree that the setting network configuration at boot time. Let me change back to bootcmd and verify it.

Reply:
Change from runcmd to bootcmd doesn't work after testing :thinking:

Reply:
Interesting, probably we are facing the same race condition we faced in libvirt that forced us to include an ugly sleep on runcmd commands for libvirt. We are missing something here I could not manage to figure out a couple of weeks ago. Anyway, note that I did not dislike to move the wicked restart at runcmd, just that I did not understand it neither expect it, probably this was a correct approach from the very beginning, it just pisses me of not understanding it what's actually happening :P

By the way, how are you testing? Are you testing on libvirt?

Reply:
No, I tested on vSphere, I think the baseOS and cloud-init behavior are the same. It should be platform independent.

But if you like, I could make a env with DHCP server in libvirt.

Reply:
Right if having those commands on runcmd works fine I am also fine with it. Do you think it would make sense to include those commands as part of `cloud-init/commands.tpl` rather than include them here directly?

Reply:
Well, put these parts into `commands.tpl` is also fine for me.
But include here is more intuitively at least to me. So personally I'd prefer to keep it in `common.tpl`.

If there is any good purpose change to `commands.tpl`, I agree to move it.

Reply:
I have no strong arguments in any direction, thus I am fine if you prefer to include them here. I also have no clear clue on why the `commands.tpl` is treated separately.

I am sorry if my review took you some extra cycles, at the end, looks like we might end up with your initial proposal. Anyway I think it is good we discussed it.

Reply:
Thanks for giving me the hint that `commands.tpl` exist :bow:, I did not notice this file before.